ProcessOptimizer/model_systems/noise_models.py

```diff
@@ -36,13 +50,19 @@ def _sample_noise(self) -> float:
         if self.noise_size is None:
             raise TypeError("Method \"raw_noise()\" for NoiseModel class "
                             f"{self.__class__.__name__} is not supposed to be called.")
-        return self._noise_distribution()*self.noise_size
+            
+        if self.noise_type == "uniform":
+            return self._noise_distribution(low=-1, high=1)*self.noise_size
+        else:
+            return self._noise_distribution()*self.noise_size
     
-    def set_noise_type(self,noise_type: str):
+    def set_noise_type(self, noise_type: str):
         if noise_type in ["normal", "Gaussian", "norm"]:
-            self._noise_distribution = norm.rvs
+            self.noise_type = noise_type
+            self._noise_distribution = self._rng.normal
         elif noise_type == "uniform":
-            self._noise_distribution = uniform(loc=-1,scale=2).rvs
+            self.noise_type = noise_type
+            self._noise_distribution = self._rng.uniform
```

Comment:
Shouldn't we have the `low=-1, high=1` here? That way, we only have one place where the code reacts to whihc distribution is used, which is going to make bug hunting easier.

```suggestion
            self._noise_distribution = lamdba: self._rng.uniform(low=-1, high=1)
```

Reply:
I don't have a strong opinion on this, since I can't think of any scenario where I would personally use this noise distribution. The main reason I kept it as the function handle was that I couldn't figure out how to code it in a way where it didn't change to a call of the function. I assume that the lambda addition fixes this.

Reply:
We could also just remove the uniform noise. It is mostly there to exemplify that you could use other noise distributions, but since we are going with numpy's default RNG, it is kind of implied anyway.

But yes, the lambda is a very succinct definition of a function.